rest-tests/src/test/java/io/syndesis/qe/upgrade/UpgradeSteps.java

```diff
@@ -1,139 +1,112 @@
 
-import org.apache.commons.lang3.StringUtils;
 import org.json.JSONArray;
 import org.json.JSONObject;
 import org.springframework.beans.factory.annotation.Autowired;
 
-import java.io.IOException;
 import java.math.BigDecimal;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 
 import cucumber.api.java.en.Then;
 import cucumber.api.java.en.When;
-import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.LocalObjectReference;
 import lombok.extern.slf4j.Slf4j;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
 
 @Slf4j
 public class UpgradeSteps {
-    private static final String VERSION_ENDPOINT = "/api/v1/version";
+    private static final String RELEASED_OPERATOR_IMAGE = "registry.redhat.io/fuse7/fuse-online-operator";
+    private static final String UPSTREAM_OPERATOR_IMAGE = "syndesis/syndesis-operator";
     private static final String DOCKER_HUB_SYNDESIS_TAGS_URL = "https://hub.docker.com/v2/repositories/syndesis/syndesis-server/tags/?page_size=1024";
-    private static final String OPERATOR_IMAGE = "syndesis/syndesis-operator:";
 
     @Autowired
     private IntegrationsEndpoint integrationsEndpoint;
 
     private String integrationId;
 
-    @When("^get upgrade versions$")
+    @When("^prepare upgrade$")
     public void getUpgradeVersions() {
-        if (System.getProperty("syndesis.upgrade.version") == null) {
-            // Parse "1.5"
-            final BigDecimal version = new BigDecimal(Double.parseDouble(StringUtils.substring(System.getProperty("syndesis.version"), 0, 3)))
-                .setScale(1, BigDecimal.ROUND_HALF_UP);
-            Request request = new Request.Builder()
-                .url(DOCKER_HUB_SYNDESIS_TAGS_URL)
-                .build();
-            String response = "";
-            try {
-                response = new OkHttpClient.Builder().build().newCall(request).execute().body().string();
-            } catch (IOException e) {
-                log.error("Unable to get version from " + DOCKER_HUB_SYNDESIS_TAGS_URL);
-                e.printStackTrace();
-            }
-
-            JSONArray jsonArray = new JSONObject(response).getJSONArray("results");
+        Syndesis syndesis = ResourceFactory.get(Syndesis.class);
+        // If it is a prod build, we can use released images from registry.redhat.io
+        if (TestUtils.isProdBuild()) {
+            final String floatingTag = TestConfiguration.syndesisOperatorImage().split(":")[1].split("-")[0];
+            final BigDecimal currentVersion = new BigDecimal(floatingTag).setScale(1, BigDecimal.ROUND_HALF_UP);
+            syndesis.setOperatorImage(RELEASED_OPERATOR_IMAGE + ":" + currentVersion.subtract(new BigDecimal("0.1")));
+            TestConfiguration.get().overrideProperty(TestConfiguration.SYNDESIS_UPGRADE_CURRENT_VERSION, TestConfiguration.syndesisVersion());
+            // Previous version needs to be specified manually via system properties
```

Comment:
shouldn't this comment be inside the `else` branch?

Reply:
no, for prod build you need to specify it manually, for upstream, it will be the same as the tag of the docker container found in docker hub